x/stake/handler.go

```diff
@@ -112,6 +112,8 @@ func handleMsgCreateValidator(ctx sdk.Context, msg types.MsgCreateValidator, k k
 	}
 
 	validator := NewValidator(msg.ValidatorAddr, msg.PubKey, msg.Description)
+	k.BumpValidatorBondHeightAndCounter(ctx, &validator)
```

Comment:
Do you think we can already set the BondHeight here?
If the maxValidators is already reached and the new validator does not have enough bonded tokens to be included in the Validator set can we still call him "Bonded". I'd say no because bond() is never called because of the EndBlocker code:

https://github.com/cosmos/cosmos-sdk/blob/663e954b806b5e8cc4206dd23d6770e7af4cfc2b/x/stake/keeper/val_state_change.go#L40

So in the end we would have a BondHeight even though the validator was never bonded / added to the voting set.

Reply:
It's OK, but semantically wrong.  And, as per #2909 (comment) I will implement a solution w/ addresses for GoS.